lib/client/weblogin.go

```diff
@@ -237,21 +237,66 @@ func SSHAgentOIDCLogin(proxyAddr, connectorID string, pubKey []byte, ttl time.Du
 		log.Debugf("got timeout waiting for callback")
 		return nil, trace.Wrap(trace.Errorf("timeout waiting for callback"))
 	}
+}
+
+// PingResponse contains data about the Teleport server like supported authentication
+// types, server version, etc.
+type PingResponse struct {
+	// Auth contains the forms of authentication the auth server supports.
+	Auth *AuthenticationSettings `json:"auth"`
```

Comment:
+1 structs are better be nested by default (reduces heap pressure, makes it easier to initialize them and removes the need to check for nils) unless there's an explicit reason to have nil in there.